src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs

```diff
@@ -336,12 +336,21 @@ public async Task<List<JObject>> ExecuteArmQueryAsync(ResourceIdentifier resourc
                 string responseContent = await response.Content.ReadAsStringAsync();
                 JObject responseObject = JObject.Parse(responseContent);
 
-                var returnedObjects = responseObject["value"].ToObject<List<JObject>>();
-                allItems.AddRange(returnedObjects);
+                string valueKey = "value";
+                IList<JObject> returnedObjects = null;
+                if (responseObject.ContainsKey(valueKey))
+                {
+                    returnedObjects = responseObject[valueKey].ToObject<List<JObject>>();
+                    allItems.AddRange(returnedObjects);
+                }
+                else
+                {
+                    allItems.Add(responseObject);
+                }
 
                 // Link to next page
                 string nextLinkToken = responseObject.GetValue("nextLink", StringComparison.InvariantCulture)?.ToString();
-                nextLink = (string.IsNullOrWhiteSpace(nextLinkToken) || !returnedObjects.Any()) ? null : new Uri(nextLinkToken);
+                nextLink = (string.IsNullOrWhiteSpace(nextLinkToken) || !(returnedObjects?.Any()).GetValueOrDefault() ? null : new Uri(nextLinkToken);
```

Comment:
>GetValueOrDefault [](start = 98, length = 17)

please put "?? false" inside the braces - it's clearer

Reply:
Agreed. Done.